### ISSUE TITLE
Crash on terraform plan when S3 bucket in config doesn't have lifecycle_rule

### ISSUE THREAD

**Issue by volodymyr-mykhailyk** [hashicorp/terraform#11719]
Hi there,

Crash occures after I imported s3 bucket, created simple config to define imported object and executed `terraform plan -refresh=false`. Crash occurs when s3 bucket state has lifecycle_rule and config doesn't

### Terraform Version
`Terraform v0.8.5`

### Affected Resource(s)
- aws_s3_bucket

### Terraform Configuration Files
```hcl
resource "aws_s3_bucket" "test_bucket" {
  bucket = "test.bucket.com"

  acl = "private"
  region = "us-west-1"
}
```
### Terraform State for s3 bucket
```json
{
  "aws_s3_bucket.mh_production": {
    "type": "aws_s3_bucket",
    "depends_on": [],
    "primary": {
      "id": "test.bucket.com",
      "attributes": {
        "acceleration_status": "",
        "arn": "arn:aws:s3:::test.bucket.com",
        "bucket": "test.bucket.com",
        "bucket_domain_name": "test.bucket.com.s3.amazonaws.com",
        "cors_rule.#": "1",
        "cors_rule.0.allowed_headers.#": "0",
        "cors_rule.0.allowed_methods.#": "1",
        "cors_rule.0.allowed_methods.0": "GET",
        "cors_rule.0.allowed_origins.#": "1",
        "cors_rule.0.allowed_origins.0": "*test.bucket.com",
        "cors_rule.0.expose_headers.#": "0",
        "cors_rule.0.max_age_seconds": "3000",
        "hosted_zone_id": "#######",
        "id": "test.bucket.com",
        "lifecycle_rule.#": "1",
        "lifecycle_rule.0.abort_incomplete_multipart_upload_days": "7",
        "lifecycle_rule.0.enabled": "true",
        "lifecycle_rule.0.expiration.#": "1",
        "lifecycle_rule.0.expiration.3591068768.date": "",
        "lifecycle_rule.0.expiration.3591068768.days": "0",
        "lifecycle_rule.0.expiration.3591068768.expired_object_delete_marker": "true",
        "lifecycle_rule.0.id": "Previous Versions cleanup",
        "lifecycle_rule.0.noncurrent_version_expiration.#": "1",
        "lifecycle_rule.0.noncurrent_version_expiration.1161702070.days": "60",
        "lifecycle_rule.0.noncurrent_version_transition.#": "1",
        "lifecycle_rule.0.noncurrent_version_transition.1377917700.days": "30",
        "lifecycle_rule.0.noncurrent_version_transition.1377917700.storage_class": "STANDARD_IA",
        "lifecycle_rule.0.prefix": "",
        "lifecycle_rule.0.transition.#": "0",
        "region": "us-west-1",
        "request_payer": "BucketOwner",
        "tags.%": "0",
        "versioning.#": "1",
        "versioning.0.enabled": "true",
        "versioning.0.mfa_delete": "false",
        "website.#": "0"
      },
      "meta": {
      },
      "tainted": false
    },
    "deposed": [],
    "provider": "aws"
  }
}
```

### Panic Output
https://gist.github.com/volodymyr-mykhailyk/63eb44bb208743177a4508926dd20e36

### Expected Behavior
Terraform showing plan with lifecycle_rule difference

### Actual Behavior
Terraform crashes

### Steps to Reproduce
1. `terraform import` - import s3 bucket with lifecycle rules
1. Add simple config for imported s3 bucket without lifecycle rules
1. `terraform plan -refresh=false` - to see the changes plan

**Comment by grubernaut** [comment]
Hi @volodymyr-mykhailyk, thanks for the issue!

This has been fixed in #11498, and will be released in the next minor version of Terraform (v0.8.6). 
Closing for now, happy to discuss further though.

**Comment by volodymyr-mykhailyk** [comment]
Thanks
